### PR TITLE
Ability to recurring expenses in the portal 

### PR DESCRIPTION
**Pull request by pankaj-ag**
This is our first cut to move in the direction of organization expenses that's why I am not linking to any issue for now. 
I have made the following changes in the PR: 
1. Setup CURD Ops for recurring expenses (Subscriptions)
2. Added a scaffolding to link the recurring expenses to all the expenses
3. A migration and modal for recurring expenses
4. Remove some unwanted and commented code from the old modules. 

### Checklist:

- [x] I have performed a self-review of my own code.

### CODE REVIEW

File: Modules/Expense/Entities/RecurringExpense.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            $month = date('m');
+
+            return Carbon::parse($initialDueDate->format("Y-$month-d"));
+        }
+
+        if ($frequency == 'yearly') {
+            $year = date('Y');
+
+            return Carbon::parse($initialDueDate->format("$year-m-d"));
+        }
```
Comment (P4NK4J) [inline Modules/Expense/Entities/RecurringExpense.php:46]: @pankaj-ag Do you think using switch case would be helpful here, we may have recurring expenses quarterly, fortnightly or weekly as well ?

File: Modules/Expense/Http/Controllers/RecurringExpenseController.php [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    }
+
+    public function create()
+    {
+        return view('expense::recurring.create', ['countries' => Country::all()]);
+    }
+
+    public function store(Request $request)
+    {
+        // ToDo:: we need to add validations here.
```
Comment (P4NK4J) [inline Modules/Expense/Http/Controllers/RecurringExpenseController.php:33]: Thanks for adding the ToDos

File: Modules/Expense/Resources/views/recurring/edit.blade.php [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+            <h4 class="mb-1 pb-1 fz-28">Edit {{ $recurringExpense->name }}</h4>
+            <span>
+                <form action={{ route('expense.recurring.destroy', $recurringExpense) }} method="POST"
+                    onsubmit="return confirm('Are you sure?')">
+                    @csrf
+                    @method('delete')
+                    <button type="submit" class="btn btn-danger text-white">Remove</button>
+                </form>
+
+
```
Comment (P4NK4J) [inline Modules/Expense/Resources/views/recurring/edit.blade.php:16]: Let's remove these spaces

**Review by P4NK4J — CHANGES_REQUESTED** [review]
Looks good. Left some comments. Also, if we can remove the extra spaces from blade files, that would be great

File: Modules/Expense/Entities/RecurringExpense.php (hunk 1 above)
Comment (pankaj-ag) [reply Modules/Expense/Entities/RecurringExpense.php:46]: @P4NK4J I think we will use a switch case when we have more than 3 conditions. I don't see a possibility of that right now.

File: Modules/Expense/Resources/views/recurring/edit.blade.php (hunk 3 above)
Comment (pankaj-ag) [reply Modules/Expense/Resources/views/recurring/edit.blade.php]: Done and sorry about that, Need to reinstall my IDE format extension.

File: Modules/Expense/Entities/RecurringExpense.php (hunk 1 above)
Comment (P4NK4J) [reply Modules/Expense/Entities/RecurringExpense.php:46]: Yeah that makes sense, I was only thinking in terms of future possibilities, but if we are only having monthly expenses as of now, I am fine with that.

**Review by P4NK4J — APPROVED** [review]
Looks good.

**Pull request merged**